### DISCUSSION TITLE
Is there any way to display the keyboard mode on screen?

### DISCUSSION THREAD

**Discussion by fahadbr** [kovidgoyal/kitty#7294] [Q&A] [answered]
I want to use modal keyboard mappings (e.g. `map --new-mode <mode> <shortcut>`) but what i would really like to have as part of this is a way to show the currently active keyboard mode on screen.  I can't seem to find a way to do this, I've searched discussions and docs with no luck on how to do this. 

Is there any way to achieve this through other means (e.g. like setting the tab title once the mode has been entered and resetting it afterwards)?

**Comment by fahadbr** [comment]
Actually i found a workaround using the `push_keyboard_mode` command. the only issue is that i have to have a separate mapping that defines the mode. Ideally i'd like to be able to combine the mapping that defines the new mode with the one that alters the title. I mapped the `--new-mode` line to a key that i'd never accidentally press for now

Here is what i have (p.s. i know there is already an interactive resize mode, but i like this way of doing it)
```
map --new-mode resize kitty_mod+f1>r 
map kitty_mod+r combine : push_keyboard_mode resize : set_tab_title RESIZE (h=narrower,l=wider,k=taller,j=shorter)

map --mode resize h resize_window narrower 4
map --mode resize l resize_window wider 4
map --mode resize k resize_window taller 4
map --mode resize j resize_window shorter 4
map --mode resize ctrl+h resize_window narrower 1
map --mode resize ctrl+l resize_window wider 1
map --mode resize ctrl+k resize_window taller 1
map --mode resize ctrl+j resize_window shorter 1
map --mode resize esc combine : pop_keyboard_mode : set_tab_title ""
map --mode resize enter combine : pop_keyboard_mode : set_tab_title ""
```

**Comment by kovidgoyal** [comment, marked as answer]
See https://github.com/kovidgoyal/kitty/commit/23779da2dc1c13ff433db59e401742ed87535f37

**Reply by fahadbr** [reply]
this is perfect, thank you for this and maintaining such a great TE!

**Reply by steakhutzeee** [reply]
Hi, could you provide an example of how you implemented this?

Thank you!

**Reply by fahadbr** [reply]
@steakhutzeee see

https://github.com/fahadbr/dotfiles/blob/master/common-home/.config/kitty/kitty.conf

line 125

```

map --new-mode resize kitty_mod+f1>r --on-unknown passthrough
map kitty_mod+r combine : push_keyboard_mode resize : set_tab_title RESIZE (h=narrower,l=wider,k=taller,j=shorter)

```

edit:
sorry this is my original implementation. i guess i never updated it to use the new tab title template

**Reply by steakhutzeee** [reply]
No prob! Thank you!

If I understand this right I should use it like:

```
# Tab mode
map --new-mode tab kitty_mod+t combine : push_keyboard_mode tab : tab_title_template "{keyboard_mode}"

[...]

# Exit tab mode
map --mode tab esc combine : pop_keyboard_mode : tab_title_template "{keyboard_mode}"
```

correct?

If yes, why you just not did:

`map --new-mode resize kitty_mod+r combine : push_keyboard_mode resize : set_tab_title RESIZE (h=narrower,l=wider,k=taller,j=shorter)` ?

**Comment by mike-lloyd03** [comment]
Is there a way to get the keyboard mode in `tab_bar.py` without passing it through as a tab title?

**Reply by kovidgoyal** [reply]
Look at the commit impleting this feature, use the same code basically something like get_boss().current_keybard_mapping.name or whatever.

**Reply by mike-lloyd03** [reply]
Thank you sir. `mode =  get_boss().mappings.current_keyboard_mode_name` did the trick
